src/utils/runtime/src/lib.rs

```diff
@@ -194,18 +198,20 @@ pub fn init_risingwave_logger(settings: LoggerSettings, registry: prometheus::Re
     // fmt layer (formatting and logging to stdout)
     {
         let fmt_layer = tracing_subscriber::fmt::layer()
-            .compact()
             .with_timer(default_timer.clone())
             .with_ansi(settings.colorful);
-        let fmt_layer = if ENABLE_PRETTY_LOG {
-            fmt_layer.pretty().boxed()
-        } else {
-            fmt_layer.boxed()
+
+        let fmt_layer = match deployment {
+            Deployment::Ci => fmt_layer
+                .compact()
+                .with_filter(FilterFn::new(|metadata| metadata.is_event())) // filter-out all span-related info
+                .boxed(),
+            Deployment::Cloud => fmt_layer.json().boxed(),
+            Deployment::Other => fmt_layer.pretty().boxed(),
```

Comment:
For local, I'm not sure if `pretty` is too verbose. 🤪

For cloud, is it the best practice to log JSON to stdout? :eyes: I imagined it's sth like writing to some log collector service directly.

Reply:
It seems scraping lines from log files and then parsing them with JSON is somehow a common practice in Loki.

- https://grafana.com/docs/loki/latest/clients/promtail/#log-file-discovery
- https://grafana.com/docs/loki/latest/clients/promtail/stages/json/